client/state/posts/test/actions.js

```diff
@@ -0,0 +1,138 @@
+	POSTS_REQUEST_FAILURE
+} from 'state/action-types';
+import {
+	receivePost,
+	receivePosts,
+	requestSitePosts
+} from '../actions';
+
+describe( 'actions', () => {
+	describe( '#receivePost()', () => {
+		it( 'should return an action object', () => {
+			const post = { ID: 841, title: 'Hello World' };
+			const action = receivePost( post );
+
+			expect( action ).to.eql( {
+				type: POST_RECEIVE,
+				post
+			} );
+		} );
+	} );
+
+	describe( '#receivePosts()', () => {
+		it( 'should return an action object', () => {
+			const posts = [ { ID: 841, title: 'Hello World' } ];
+			const action = receivePosts( posts );
+
+			expect( action ).to.eql( {
+				type: POSTS_RECEIVE,
+				posts
+			} );
+		} );
+	} );
+
+	describe( '#requestSitePosts()', () => {
+		const spy = sinon.spy();
+
+		before( () => {
+			Chai.use( sinonChai );
+
+			nock( 'https://public-api.wordpress.com:443' )
+				.persist()
+				.get( '/rest/v1.1/sites/2916284/posts' )
+				.reply( 200, {
+					posts: [
+						{ ID: 841, title: 'Hello World' },
+						{ ID: 413, title: 'Ribs & Chicken' }
+					]
+				} )
+				.get( '/rest/v1.1/sites/2916284/posts' )
+				.query( { search: 'Hello' } )
+				.reply( 200, {
+					posts: [ { ID: 841, title: 'Hello World' } ]
+				} )
+				.get( '/rest/v1.1/sites/77203074/posts' )
+				.reply( 403, {
+					error: 'authorization_required',
+					message: 'User cannot access this private blog.'
+				} );
+		} );
+
+		beforeEach( () => {
+			spy.reset();
+		} );
+
+		after( () => {
+			nock.restore();
+		} );
+
+		it( 'should dispatch fetch action when thunk triggered', () => {
+			requestSitePosts( 2916284 )( spy );
+
+			expect( spy ).to.have.been.calledWith( {
+				type: POSTS_REQUEST,
+				siteId: 2916284,
+				query: {}
+			} );
+		} );
+
+		it( 'should dispatch posts receive action when request completes', ( done ) => {
+			requestSitePosts( 2916284 )( spy ).then( () => {
+				expect( spy ).to.have.been.calledWith( {
+					type: POSTS_RECEIVE,
+					posts: [
+						{ ID: 841, title: 'Hello World' },
+						{ ID: 413, title: 'Ribs & Chicken' }
+					]
+				} );
+
+				done();
+			} ).catch( done );
+		} );
+
+		it( 'should dispatch posts posts request success action when request completes', ( done ) => {
+			requestSitePosts( 2916284 )( spy ).then( () => {
+				expect( spy ).to.have.been.calledWith( {
+					type: POSTS_REQUEST_SUCCESS,
+					siteId: 2916284,
+					query: {},
+					posts: [
+						{ ID: 841, title: 'Hello World' },
+						{ ID: 413, title: 'Ribs & Chicken' }
+					]
+				} );
+
+				done();
+			} ).catch( done );
+		} );
+
+		it( 'should dispatch fail action when request fails', ( done ) => {
+			requestSitePosts( 77203074 )( spy ).then( () => {
+				expect( spy ).to.have.been.calledWith( {
+					type: POSTS_REQUEST_FAILURE,
+					siteId: 77203074,
+					query: {},
+					error: sinon.match( { message: 'User cannot access this private blog.' } )
+				} );
+
+				done();
+			} ).catch( done );
+		} );
+	} );
+} );
```

Comment:
>this is a great example test file and is worth publishing in a weekly digest

Thanks @dmsnell !

Related, I had struggled for a while to find a way to test the dispatch callback without relying on the return value being a promise. I think this is reflected by the requested Sinon feature at sinonjs/sinon#118, as it could allow me to write something like:

```jsx
// Before:

requestSitePosts( 77203074 )( spy ).then( () => {
	expect( spy ).to.have.been.calledWith( {
		type: POSTS_REQUEST_FAILURE,
		siteId: 77203074,
		query: {},
		error: sinon.match( { message: 'User cannot access this private blog.' } )
	} );
	done();
} ).catch( done );

// After:

const stub = sinon.stub().calledWith( {
	type: POSTS_REQUEST_FAILURE,
	siteId: 77203074,
	query: {},
	error: sinon.match( { message: 'User cannot access this private blog.' } )
} ).calls( done );

requestSitePosts( 77203074 )( stub );
```

Curious if anyone has encountered a similar need.